irc/modes/modes.go

```diff
@@ -291,6 +291,23 @@ func ParseChannelModeChanges(params ...string) (ModeChanges, map[rune]bool) {
 						continue
 					}
 				}
+			case Key:
+				// #874: +k is technically a type B mode, requiring a parameter
+				// both for add and remove. so attempt to consume a parameter,
+				// but allow remove (but not add) even if no parameter is available.
+				// however, the remove parameter should always display as "*", matching
+				// the freenode behavior.
+				if len(params) > skipArgs {
+					if change.Op == Add {
+						change.Arg = params[skipArgs]
+					}
+					skipArgs++
+				} else if change.Op == Add {
+					continue
+				}
+				if change.Op == Remove {
+					change.Arg = "*"
+				}
```

Comment:
This way could show the comment's intent clearer (forbid `+k` with no param, `-k` argument is always `*`)

```suggestion
				if len(params) <= skipArgs && change.Op == Add {
					continue
				}
				if change.Op == Add {
					change.Arg = params[skipArgs]
				} else if change.Op == Remove {
					change.Arg = "*"
				}
				if len(params) > skipArgs {
					skipArgs++
				}
```